R/ccfr_uncertainty.R

```diff
@@ -0,0 +1,74 @@
+#' Corrected case fatality rate
+#'
+#' @description Calculates the maximum likelihood estimate and 95% confidence
+#' interval of a corrected CFR, using the total cases and total cases with known
+#' outcomes, where the latter replaces the total number of deaths in the
+#' standard (naive) CFR definition. We use a binomial likelihood, approximated
+#' by a Poisson likelihood for large samples
+#'
+#' @param total_cases The total number of cases observed over the period of an
+#' outbreak of interest. The total number of cases must be greater than or equal
+#' to the total number of deaths
+#'
+#' @param total_deaths The total number of deaths observed over the period of an
+#' outbreak of interest. The total number of deaths must be less than or equal
+#' to the total number of cases
+#'
+#' @param u_t The proportion of cases to cases with known outcomes up to the
+#' point of an outbreak of interest. Used to correct the total number of deaths
+#' for delays between case detection and outcome. Given that its a proportion,
+#' it must be between 0.0 and 1.0.
+#' @param poisson_threshold The case count above which to use Poisson
+#' approximation. Set to 200 by default.
+#'
+#' @return A named vector with the MLE and 95% confidence interval of the
+#' corrected CFR estimates, named "cfr_me", "cfr_low", and "cfr_high".
+#' @export
+#'
+#' @examples
+#' ccfr_uncertainty(total_cases = 1000, total_deaths = 900, u_t = 0.2)
+ccfr_uncertainty <- function(total_cases,
+                             total_deaths,
+                             u_t,
+                             poisson_threshold = 200) {
+  stopifnot(
+    "`total_cases` must be equal to or more than `total_deaths`" =
+      (total_cases >= total_deaths)
+  )
+
+  # MLE estimation for corrected CFR
```

Comment:
It would be good to define the likelihood estimation function outside of the loop and avoid the repeated if/else and improve performance while making things more readable

Reply:
Changed to external definition and using a `vapply` in https://github.com/epiverse-trace/datadelay/pull/11/commits/74af9360e3c410d63b971c81235547872df95f53

Reply:
Vectorised instead in https://github.com/epiverse-trace/datadelay/pull/11/commits/8cb1987e3a3164b239aa86ef4ffd4c57ac6e2e6a